app/controllers/users_controller.rb

```diff
@@ -22,8 +30,50 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.update_attributes user_params
+      flash[:success] = t ".prupdate"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @user = User.find_by id: params[:id]
```

Comment:
Will the destroy work if @user is nil? 
Can you try using "before_action" filter here. Study about filter in rails. 
Reference: [http://guides.rubyonrails.org/action_controller_overview.html#filters](http://guides.rubyonrails.org/action_controller_overview.html#filters)

Reply:
Yes